### PR TITLE
Chain page tweaks

### PR DESCRIPTION
**Pull request by MananTank**
## PR-Codex overview
This PR enhances chain-related components in the dashboard app. 

### Detailed summary
- Updated conditional rendering in `page.tsx` for `ChainCTA`
- Improved styling and behavior in `sidebar-content.tsx`
- Added new components like `MobileMenu`, `cta-card.tsx`
- Updated metadata generation behavior in `layout.tsx`
- Enhanced wallet connection functionality in `add-chain-to-wallet.tsx`
- Added `useCustomConnectModal` hook in `index.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/@/components/ui/tooltip.tsx [hunk 1]
```diff
@@ -31,14 +31,20 @@ export { Tooltip, TooltipTrigger, TooltipContent, TooltipProvider };
 
 export function ToolTipLabel(props: {
   children: React.ReactNode;
-  label: string;
+  label: string | undefined;
   rightIcon?: React.ReactNode;
   leftIcon?: React.ReactNode;
 }) {
+  if (!props.label) {
+    return <>{props.children}</>;
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/ui/tooltip.tsx:39]: don't think you need the fragment here? Newer react types handle just returning props.children fine I believe?

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/MobileMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        </Button>
+      </DialogTrigger>
+      <DialogContent
+        className="p-0"
+        dialogCloseClassName="hidden"
+        onClick={(e) => {
+          if (e.target instanceof HTMLAnchorElement) {
+            setIsOpen(false);
+          }
+        }}
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/MobileMenu.tsx:28]: oof I do not like this indirection but I understand why its needed

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/add-chain-to-wallet.tsx [hunk 3]
```diff
@@ -12,58 +11,58 @@ import {
   useSwitchActiveWalletChain,
 } from "thirdweb/react";
 import { useDebounce } from "use-debounce";
+import { ToolTipLabel } from "../../../../../../../@/components/ui/tooltip";
+import { useCustomConnectModal } from "../../../../../../../@3rdweb-sdk/react/components/connect-wallet";
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/add-chain-to-wallet.tsx:15]: use absolute paths here

**Review by jnsdls — CHANGES_REQUESTED** [review]
mostly nits

File: apps/dashboard/src/@/components/ui/tooltip.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/ui/tooltip.tsx:39]: removed

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/add-chain-to-wallet.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/add-chain-to-wallet.tsx:15]: Added vscode setting to automatically do this

**Review by jnsdls — APPROVED** [review]

**Pull request merged**